src/nano_contracts/on_chain_blueprint.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable max-classes-per-file */
+
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import zlib from 'zlib';
+import { ON_CHAIN_BLUEPRINTS_INFO_VERSION, ON_CHAIN_BLUEPRINTS_VERSION } from '../constants';
+import Transaction from '../models/transaction';
+import { intToBytes } from '../utils/buffer';
+
+export enum CodeKind {
+  PYTHON_GZIP = 'python+gzip',
+}
+
+export class Code {
+  kind: CodeKind;
+
+  content: Buffer;
+
+  constructor(kind: CodeKind, content: Buffer) {
+    this.kind = kind;
+    this.content = content;
+  }
+
+  serialize(): Buffer {
+    const arr: Buffer[] = [];
+    if (this.kind !== CodeKind.PYTHON_GZIP) {
+      throw new Error('Invalid code kind value');
+    }
+
+    const zcode = zlib.deflateSync(this.content);
+    arr.push(Buffer.from(this.kind, 'utf8'));
+    arr.push(intToBytes(0, 1));
+    arr.push(zcode);
+    return Buffer.concat(arr);
+  }
+}
+
+class OnChainBlueprint extends Transaction {
```

Comment:
suggestion(non-blocking): Add a class docstring here to explain any specifics of the `OnChainBlueprint` class.

For instance, it seems that this tx has no inputs or outputs and no token array either, maybe we should explain why.